File: 917. Reverse Only Letters.py
# PYTHON
#* 917. Reverse Only Letters
s = "ab-cd"
def reverse_only_letters(s):
    # Collect all the letters in the string
    letters = [c for c in s if c.isalpha()]
    # Replace letters in the string with those from the collected letters in reverse order
    result = ''.join([letters.pop() if c.isalpha() else c for c in s])
    return result

print(reverse_only_letters("ab-cd"))          # Output: "dc-ba"
print(reverse_only_letters("a-bC-dEf-ghIj"))  # Output: "j-Ih-gfE-dCba"
print(reverse_only_letters("Test1ng-Leet=code-Q!"))  # Output: "Qedo1ct-eeLg=ntse-T!"

File: 917. Reverse Only Letters.js
// PYTHON
// 917. Reverse Only Letters
s = "ab-cd"
/**
 * @param {string} s
 * @return {string}
 */
//SOL
var reverseOnlyLetters = function(s) {
    // Collect all the letters in the string
    let letters = s.split('').filter(c => /[a-zA-Z]/.test(c));
    // Replace letters in the string with those from the collected letters in reverse order
    let result = s.split('').map(c => /[a-zA-Z]/.test(c) ? letters.pop() : c).join('');
    return result;
}
console.log(reverseOnlyLetters("ab-cd"));          // Output: "dc-ba"
console.log(reverseOnlyLetters("a-bC-dEf-ghIj"));  // Output: "j-Ih-gfE-dCba"
console.log(reverseOnlyLetters("Test1ng-Leet=code-Q!"));  // Output: "Qedo1ct-eeLg=ntse-T!"